src/main/java/de/tum/in/www1/artemis/domain/Exercise.java

```diff
@@ -921,4 +922,60 @@ public String getMappedColumnName() {
         }
     }
 
+    /**
+     * This method is used to validate the dates of an exercise. A date is valid if there is no dueDateError or assessmentDueDateError
+     * @throws BadRequestException if the dates are not valid
+     */
+    public void validateDates() throws BadRequestException {
+        // All fields are optional, so there is no error if none of them is set
+        if (getReleaseDate() == null && getDueDate() == null && getAssessmentDueDate() == null) {
+            return;
+        }
+        // atleast one is set, so we have to check the two possible errors
+        boolean validDates = validateDueDateError(getReleaseDate(), getDueDate(), getAssessmentDueDate())
+                && validateAssessmentDueDateError(getReleaseDate(), getDueDate(), getAssessmentDueDate());
+
+        if (!validDates) {
+            throw new BadRequestException();
+        }
+    }
+
+    /**
+     * This method is used to validate the dueDate of an exercise. A due date is valid, if it is after the releaseDate,
+     * @return true if there is no DueDateError
+     */
+
+    private boolean validateDueDateError(ZonedDateTime releaseDate, ZonedDateTime dueDate, ZonedDateTime assesmentDueDate) {
+        if (releaseDate == null) {
+            return true;
+        }
+        return validateTwoDates(releaseDate, dueDate);
+    }
+
+    /**
+     * This method is used to validate the assesmentDueDate of an exercise. An assessmentDueDate is valid if it is after the releaseDate and dueDate. A given assesmentDueDate is invalid without an according dueDate
+     * @return true if there is no assessmentDueDateError
+     */
+    private boolean validateAssessmentDueDateError(ZonedDateTime releaseDate, ZonedDateTime dueDate, ZonedDateTime assessmentDueDate) {
+        if (assessmentDueDate == null) {
+            return true;
+        }
+        // There cannot be a assessmentDueDate without dueDate
+        if (dueDate == null) {
+            return false;
+        }
+        return validateTwoDates(dueDate, assessmentDueDate) && validateTwoDates(releaseDate, assessmentDueDate);
+    }
+
+    /**
+     * This method is used to validate if the previousDate is before the laterDate.
+     * @return true if the previousDate is valid
+     */
+    private boolean validateTwoDates(ZonedDateTime previousDate, ZonedDateTime laterDate) {
```

Comment:
```suggestion
    private boolean isBeforeAndNotNull(ZonedDateTime previousDate, ZonedDateTime laterDate) {
```
would imho be more descriptive of what the method does.